sm_timing/include/sm/timing/implementation/TimestampCorrector.hpp

```diff
@@ -67,70 +61,72 @@ typename TimestampCorrector<T>::time_t TimestampCorrector<T>::correctTimestamp(c
-  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2, "The timestamp correction requires at least two data points before this funciton can be called");
+  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2u,
+               "The timestamp correction requires at least two data points "
+               "before this funciton can be called");
 
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1u];
 
-  // look up the local timestamp
-  return  double(l2.y - l1.y) / double(l2.x - l1.x); 
+  // Look up the local timestamp.
+  return  double(l2.y - l1.y) / double(l2.x - l1.x);
 }
 
 template<typename T>
 double TimestampCorrector<T>::getOffset() const {
-  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2, "The timestamp correction requires at least two data points before this funciton can be called");
+  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2u,
+               "The timestamp correction requires at least two data points "
+               "before this function can be called");
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1u];
   
-  // look up the local timestamp
-  return  double(l1.y) + (double(-l1.x) * double(l2.y - l1.y) / double(l2.x - l1.x) ); 
+  // Look up the local timestamp.
+  return  double(l1.y) + (double(-l1.x) * double(l2.y - l1.y) / double(l2.x - l1.x) );
 }
 
   
 // Get the local time from the remote time.
 template<typename T>
-typename TimestampCorrector<T>::time_t TimestampCorrector<T>::getLocalTime(const time_t & remoteTime) const
-{
-  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2, "The timestamp correction requires at least two data points before this funciton can be called");
+typename TimestampCorrector<T>::time_t TimestampCorrector<T>::getLocalTime(
+    const time_t& remoteTime) const {
+  SM_ASSERT_GE(NotInitializedException, _convexHull.size(), 2u,
+               "The timestamp correction requires at least two data "
+               "points before this funciton can be called");
 
   // Get the line at the time midpoint.
-  const Point & l1 = _convexHull[_midpointSegmentIndex];
-  const Point & l2 = _convexHull[_midpointSegmentIndex + 1];
+  const Point& l1 = _convexHull[_midpointSegmentIndex];
+  const Point& l2 = _convexHull[_midpointSegmentIndex + 1];
 
-  // look up the local timestamp
-  return  l1.y + ((remoteTime - l1.x) * (l2.y - l1.y) / (l2.x - l1.x) ); 
-}
+  // Look up the local timestamp.
+  SM_ASSERT_GT(NotInitializedException, l2.x - l1.x, 0u, "Division by zero not allowed.");
```

Comment:
IMO, for unsigned and general times this would be better after  `l2.x - l1.x, 0u` -> `l2.x, l1.x,` . Actually I doubt that this can ever be violated because of : https://github.com/ethz-asl/Schweizer-Messer/pull/178/files#diff-9df708da0b9c55ac142ff1cbf8424a63R16.